### ISSUE TITLE
socket:read() does not read from the system socket when 0 timeout used

### ISSUE THREAD

**Issue by RunsFor** [tarantool/tarantool#3846]
**Tarantool version:**

```
Tarantool 1.10.1-370-gac77418fc
Target: Darwin-x86_64-Debug
Build options: cmake . -DCMAKE_INSTALL_PREFIX=/usr/local -DENABLE_BACKTRACE=ON
Compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++
C_FLAGS: -Wno-unknown-pragmas -fexceptions -funwind-tables -fno-omit-frame-pointer -fno-stack-protector -fno-common -msse2 -std=c11 -Wall -Wextra -Wno-strict-aliasing -Wno-char-subscripts -Werror
CXX_FLAGS: -Wno-unknown-pragmas -fexceptions -funwind-tables -fno-omit-frame-pointer -fno-stack-protector -fno-common -msse2 -std=c++11 -Wall -Wextra -Wno-strict-aliasing -Wno-char-subscripts -Werror
```

**OS version:** MacOS Mojave 10.14.1 (18B75)

**Bug description:**

When execute :read with 0 timeout (`socket:read(<size>, 0)`) nothing is read from the system socket. It only read bytes from internal buffer if any.

**Steps to reproduce:**

```
tarantool> socket = require'socket'
---
...

tarantool> fiber = require'fiber'
---
...
tarantool> socket.tcp_server('0.0.0.0', 3333, function(s) while true do s:write('1234'); fiber.sleep(10); end end)
started
---
- fd: 6
  name:
    host: 0.0.0.0
    family: AF_INET
    type: SOCK_STREAM
    protocol: tcp
    port: 3333
- host: 0.0.0.0
  family: AF_INET
  type: SOCK_STREAM
  protocol: tcp
  port: 3333
...
tarantool> sock = socket.tcp_connect('localhost', 3333)
---
...

tarantool> sock:read(4,0)
---
- null
...

tarantool> sock:read(2)
---
- '12'
...

tarantool> sock:read(2,0)
---
- '34'
...
```

**Comment by sergepetrenko** [comment]
@RunsFor couldn't reproduce. Does the issue still occur on your side?
```
Tarantool 1.10.3-55-gaa4b10212
Target: Darwin-x86_64-Debug
```
Here's my output:
```
tarantool> socket = require'socket'
---
...

tarantool> fiber = require'fiber'
---
...

tarantool> socket.tcp_server('0.0.0.0', 3333, function(s) while true do s:write('1234'); fiber.sleep(10); end end)
started
---
- fd: 6
  name:
    host: 0.0.0.0
    family: AF_INET
    type: SOCK_STREAM
    protocol: tcp
    port: 3333
- host: 0.0.0.0
  family: AF_INET
  type: SOCK_STREAM
  protocol: tcp
  port: 3333
...

tarantool> sock = socket.tcp_connect('localhost', 3333)
---
...

tarantool> sock:read(4,0)
---
- '1234'
...

```

**Comment by sergepetrenko** [comment]
Looks like the issue was resolved in commit e6bd7748b7. I could reproduce it after reverting the commit.
So you need `Tarantool 1.10.1-426-ge6bd7748b` or newer
